### PR TITLE
Fix/improve walker population carry over handling in batched drivers

### LINKED ISSUE #4436: Batched driver walker count at restart
**Describe the bug**
In batched drivers, xml input takes precedence over walker count from the previous QMC driver section or restart.
Even when no walker control input is given, input parameters default values are still used.
For this is reason, there is no way to retain the exact population.

My fix suggestion is retaining the exact population if no walker control input is given in the QMC driver section input.
Namely
````
<qmc>
  <parameter name="walkers_per_rank"    >    1190            </parameter>
</qmc>
```
or
```
<qmc>
  <parameter name="total_walkers"    >    2380            </parameter>
</qmc>
```
the previous population will be tailored based the request in the current input block.
If neither control is given in the input. The previous population is retained.

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Close #4436
```
If neither ``total_walkers`` nor ``walkers_per_rank`` is provided and there are walker configurations carried over from previuos QMC sections or a restart, the population carried over will be used without modification.
```

## What type(s) of changes does this code introduce?
- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: docs/methods.rst [hunk 1]
```diff
@@ -358,13 +358,21 @@ Additional information:
...
-  least the number of OpenMP threads for pure CPU runs. For GPU runs, a scan of this parameter is necessary to reach reasonable single rank
-  efficiency and also get a balanced time to solution.
-  If neither ``total_walkers`` nor ``walkers_per_rank`` is provided, ``walkers_per_rank`` is set equal to ``crowds``.
+- ``walkers_per_rank`` The number of walkers per MPI rank. This number does not have to be a multiple of the number of OpenMP
+  threads. However, to avoid any idle resources, it is recommended to be at least the number of OpenMP threads for pure CPU runs.
+  For GPU runs, a scan of this parameter is necessary to reach reasonable single rank efficiency and also get a balanced time to
+  solution. For highest throughput on GPUs, expect to use hundreds of walkers_per_rank, or the largest number that will fit in GPU
+  memory.
+
+  If neither ``total_walkers`` nor ``walkers_per_rank`` is provided and there are walker configurations carried over from previuos QMC sections or a restart,
```
Comment (markdewing) [inline docs/methods.rst:367]: Spelling of 'previous'.  Also in line 1751.

Comment (prckent) [reply docs/methods.rst]: I'll push a fix

**Comment by prckent** [comment]
Thanks Ye. Hopefully this works for everyone.

**Review by prckent — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

File: docs/methods.rst (hunk 1 above)
Comment (ye-luo) [reply docs/methods.rst]: Thanks.

**Pull request merged**